apps/api/src/services/agent-manager.service.ts

```diff
@@ -753,4 +758,54 @@ export class AgentManager {
       return { agents: [], total: 0 };
     }
   }
+
+  /**
+   * Get agents with paging and filtering
+   */
+  async getAgents({
+    filter,
+    pagingParams,
+  }: {
+    filter?: string;
+    pagingParams: PagingParams;
+  }) {
+    if (filter?.length === 42) {
+      return findByWallet({ walletAddress: filter, pagingParams });
+    }
+    if (typeof filter === "string" && filter.length > 0) {
+      return findByName({ name: filter, pagingParams });
+    }
+
+    return findAll(pagingParams);
+  }
+
+  /**
+   * Count agents with optional filter
+   */
+  async countAgents(filter?: string) {
+    if (filter?.length === 42) {
+      return countByWallet(filter);
+    }
+    if (filter?.length) {
+      return countByName(filter);
+    }
+
+    return count();
+  }
+
+  sanitizeAgent(agent: SelectAgent) {
```

Comment:
@dtbuchholz I added this method to handle all of the places we are sanitizing agents.  There was a few and they might as well all return the same shaped Object.